snapshot/storage/metastore.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/docker/containerd/snapshot"
+)
+
+// MetaStore is used to store metadata related to a snapshot driver. The
+// MetaStore is intended to store metadata related to name, state and
+// parentage. Using the MetaStore is not required to implement a snapshot
+// driver but can be used to handle the persistence and transactional
+// complexities of a driver implementation.
+type MetaStore interface {
+	// TransactionContext creates a new transaction context.
+	TransactionContext(ctx context.Context, writable bool) (context.Context, Transactor, error)
```

Comment:
Is a `Transactor` required for performing read-only operations like `Stat`?

(based on looking at the code for `boltMetastore` I think the answer is yes, but it'd be good to capture that requirement in the godoc here)